x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CrossClusterEsqlRCS2UnavailableRemotesIT.java

```diff
@@ -0,0 +1,310 @@
+        assertThat((int) clusters.get("running"), is(0));
+        assertThat((int) clusters.get("skipped"), is(0));
+        assertThat((int) clusters.get("partial"), is(0));
+        assertThat((int) clusters.get("failed"), is(0));
+
+        assertThat(clusterDetails.size(), is(2));
+        assertThat((int) localClusterDetails.get("took"), greaterThan(0));
+        assertThat(localClusterDetails.get("status"), is("successful"));
+
+        assertThat((int) remoteClusterDetails.get("took"), greaterThan(0));
+        assertThat(remoteClusterDetails.get("status"), is("successful"));
+    }
+
+    @SuppressWarnings("unchecked")
+    private void remoteClusterShutdownWithSkipUnavailableTrue() throws Exception {
+        // Remote cluster is stopped and skip unavailable is set to true.
+        // We expect no exception and partial results from the remaining open cluster.
+
+        configureRemoteCluster("my_remote_cluster", fulfillingCluster, false, randomBoolean(), true);
+
+        try {
+            // Stop remote cluster.
+            fulfillingCluster.stop(true);
+
+            // A simple query that targets our remote cluster.
+            String query = "FROM *,my_remote_cluster:* | LIMIT 10";
+            Response response = performRequestWithRemoteSearchUser(esqlRequest(query));
+
+            Map<String, Object> map = responseAsMap(response);
+            ArrayList<String> columns = (ArrayList<String>) map.get("columns");
+            ArrayList<String> values = (ArrayList<String>) map.get("values");
+            Map<String, ?> clusters = (Map<String, ?>) map.get("_clusters");
+            Map<String, ?> clusterDetails = (Map<String, ?>) clusters.get("details");
+            Map<String, ?> localClusterDetails = (Map<String, ?>) clusterDetails.get("(local)");
+            Map<String, ?> remoteClusterDetails = (Map<String, ?>) clusterDetails.get("my_remote_cluster");
+
+            // Assert results obtained from the local cluster and that remote cluster was
+            // skipped.
+            assertOK(response);
+            assertThat((int) map.get("took"), greaterThan(0));
+
+            assertThat(columns.size(), is(2));
+            assertThat(values.size(), is(5));
+
+            assertThat((int) clusters.get("total"), is(2));
+            assertThat((int) clusters.get("successful"), is(1));
+            assertThat((int) clusters.get("skipped"), is(1));
+            assertThat((int) clusters.get("running"), is(0));
+            assertThat((int) clusters.get("partial"), is(0));
+            assertThat((int) clusters.get("failed"), is(0));
+
+            assertThat(clusterDetails.size(), is(2));
+            assertThat((int) localClusterDetails.get("took"), greaterThan(0));
+            assertThat(localClusterDetails.get("status"), is("successful"));
+
+            assertThat((int) remoteClusterDetails.get("took"), greaterThan(0));
+            assertThat(remoteClusterDetails.get("status"), is("skipped"));
```

Comment:
It's important to look at the failures list for the skipped remote cluster. Ideally that error should be something like "could not connect to remotex", but it might be a security exception. We want to ensure that the exception is useful to the end user.